tests/Unit/App/Action/Backend/Account/UpdateAccountActionTest.php

```diff
@@ -188,6 +188,59 @@ function executeUpdateAccount(
         ->pluck('externalIdentifier')
         ->all();
 
-    // If custom PostNL carrier is present, all other PostNL entries should be removed
+    // If multiple PostNL carriers are present, the custom contract should be used.
     expect($externalIdentifiers)->toBe(['postnl:23991']);
 });
+
+it('maps carriers correctly with multiple non-contract postnl entries', function () {
+    executeUpdateAccount(['apiKey' => 'test-api-key'], null, [
+        [
+            'id'         => 1,
+            'label'      => null,
+            'carrier_id' => 1,
+            'carrier'    => [
+                'id'   => 1,
+                'name' => 'postnl',
+            ],
+            'enabled'    => 1,
+            'optional'   => 1,
+            'primary'    => 1,
+            'type'       => 'main',
+        ],
+        [
+            'id'         => 1,
+            'label'      => 'postnl_package_small_nl',
+            'carrier_id' => 1,
+            'carrier'    => [
+                'id'   => 1,
+                'name' => 'postnl',
+            ],
+            'enabled'    => 1,
+            'optional'   => 0,
+            'primary'    => 1,
+            'type'       => 'main',
+        ],
+        [
+            'id'         => 1,
+            'label'      => 'postnl_flespakket_on_myparcel',
+            'carrier_id' => 1,
+            'carrier'    => [
+                'id'   => 1,
+                'name' => 'postnl',
+            ],
+            'enabled'    => 0,
+            'optional'   => 1,
+            'primary'    => 1,
+            'type'       => 'main',
+        ],
+    ]);
+
+    $firstShop = AccountSettings::getAccount()->shops->first();
+
+    $externalIdentifiers = $firstShop->carriers
+        ->pluck('externalIdentifier')
+        ->all();
+
+    // If multiple PostNL carriers are present, but no custom contract, only the first one should be kept.
+    expect($externalIdentifiers)->toBe(['postnl']);
```

Comment:
Ziet er goed uit! Wordt er ook al getest met situaties dat je meerdere postnl carriers hebt + nog 1 of meer andere carriers? Op regel 102 zie ik wel iets dergelijks maar voor de zekerheid vraag ik het even na.

Reply:
niet per se, maar dat heeft geen impact op elkaar, echter zal ik het alsnog ff toevoegen zodat het compleet is